porcupine/plugins/filemanager.py

```diff
@@ -122,6 +157,34 @@ def rename(old_path: Path) -> None:
         tab.path = new_path / tab.path.relative_to(old_path)
 
 
+def paste(new_path: Path) -> None:
+    paste_here(new_path.parent)
+
+
+def paste_here(new_path: Path) -> None:
+    assert copy_path is not None
+    assert new_path.is_dir()
+
+    new_file_path = new_path / copy_path.name
+
+    if new_file_path.exists():
+        new_file_path = ask_name_for_renaming(new_file_path, is_paste=True)  # rename-overwrite file conflict
+        if new_file_path is None:
+            return
```

Comment:
This currently creates a mypy error. Because `ask_name_for_renaming` can return `None` but `new_file_path` has type `Path` (as it was initially set to `new_path / copy_path.name`), mypy complains that `new_file_path` can become `None`, even though it is only temporarily `None` and the rest of the code can still assume it is `Path`. We could report this to mypy's issue tracker, but it is likely that someone has already reported, and even more likely that nobody would fix it (there's lots of open issues).

Here is a workaround:

```suggestion
        path = ask_name_for_renaming(new_file_path, is_paste=True)  # rename-overwrite file conflict
        if path is None:
            return
        new_file_path = path
```

Now `path` will have type `Path | None` before the `return` statement, but because we check for `None`, it has type `Path` after the `return` statement and we can assign it to `new_file_path`.